stix_shifter/stix_transmission/src/modules/carbonblack/carbonblack_connector.py

```diff
@@ -0,0 +1,62 @@
+from ..base.base_connector import BaseConnector
+from .carbonblack_api_client import APIClient
+import json
+
+
+class Connector(BaseConnector):
+    def __init__(self, connection, configuration, dialect="process"):
+        if dialect not in ["process", "binary"]:
+            raise NotImplementedError("Invalid module dialect")
+
+        self.api_client = APIClient(connection, configuration, dialect=dialect)
+        self.ping_connector = self
+        self.results_connector = self
+        self.query_connector = self
+        self.is_async = False
+
+    def ping(self):
+        try:
+            response = self.api_client.ping_box()
+            response_code = response.code
+            response_json = json.loads(response.read())
+
+            return_obj = dict()
+
+            if len(response_json) > 0 and 200 <= response_code < 300:
+                return_obj['success'] = True
+            else:
+                return_obj['success'] = False
+                return_obj['error'] = 'error when pinging data source'
+            return return_obj
+        except Exception as err:
+            return_obj = dict()
+            return_obj['success'] = False
+            return_obj['error'] = 'error when pinging data source: {}'.format(err)
+            return return_obj
+
+    def create_query_connection(self, query):
+        return {"success": True, "search_id": query}
+
+    def create_results_connection(self, search_id, offset, length):
+        try:
+            query = search_id
+            response = self.api_client.run_search(query)
```

Comment:
is there a way to fetch particular data range? the offset/length parameters

Reply:
Yes. It supports a start and a rows parameter. I'll add that. That also made me realize that the api defaults to returning only 10 rows.